src/lib/array.jl

```diff
@@ -3,22 +3,17 @@ using FillArrays: AbstractFill, getindex_value
 using Base.Broadcast: broadcasted, broadcast_shape
 using Distributed: pmap, AbstractWorkerPool
 
-@adjoint (::Type{T})(::UndefInitializer, args...) where T<:Array = T(undef, args...), Δ -> nothing
-
 @adjoint Array(xs::AbstractArray) = Array(xs), ȳ -> (ȳ,)
 @adjoint Array(xs::Array) = Array(xs), ȳ -> (ȳ,)
 
 @nograd ones, zeros, Base.OneTo, Colon(), one, zero, sizehint!, count
 
-@adjoint Base.vect(xs...) = Base.vect(xs...), Δ -> (Δ...,)
-
 @adjoint copy(x::AbstractArray) = copy(x), ȳ -> (ȳ,)
 
 @adjoint collect(x::Tuple) = collect(x), dy -> (Tuple(dy),)
 @adjoint collect(x::AbstractArray) = collect(x), dy -> (dy,)
 
 # Array Constructors
-@adjoint (::Type{T})(x::T) where T<:Array = T(x), ȳ -> (ȳ,)
```

Comment:
Do we have an equivalent in CR?

Reply:
Added in 1.5: https://github.com/JuliaDiff/ChainRules.jl/commit/7593339839832ffb7ac83401918390ffe9d4eb42#diff-7a31ac47520a0ad235f15cead3f4c48fcb595dc7f2421ebbee8ffce23e2510f5R7